pxtrunner/renderer.ts

```diff
@@ -219,24 +219,41 @@ namespace pxt.runner {
         render: (container: JQuery, r: pxt.runner.DecompileResult) => void;
     }[] = [];
     function consumeRenderQueueAsync(): Promise<void> {
-        const job = renderQueue.shift();
-        if (!job) return Promise.resolve(); // done
-
-        const { el, options, render } = job;
-        return pxt.runner.decompileToBlocksAsync(el.text(), options)
-            .then((r) => {
-                const errors = r.compileJS && r.compileJS.diagnostics && r.compileJS.diagnostics.filter(d => d.category == pxtc.DiagnosticCategory.Error);
-                if (errors && errors.length)
-                    errors.forEach(diag => pxt.reportError("docs.decompile", "" + diag.messageText, { "code": diag.code + "" }));
-                render(el, r);
-                el.removeClass("lang-shadow");
-                return consumeRenderQueueAsync();
-            }).catch(e => {
-                pxt.reportException(e);
-                el.append($('<div/>').addClass("ui segment warning").text(e.message));
-                el.removeClass("lang-shadow");
-                return consumeRenderQueueAsync();
+        const existingFilters: Map<Element> = {};
```

Comment:
do you need to hold a ref to that Element? it is rooting it.

Reply:
I think I'm missing something here; this exists only in the scope of ``consumeRenderQueueAsync``, and will be gc-ed after all blocks are rendered. No specific reason to ref the element itself though I suppose as I really just need a set, can just do `Map<boolean>`